server/app/models/sync.rb

```diff
@@ -147,10 +147,14 @@ def perform_post_discard_sync
   def stream_name_exists?
     return if destination.blank?
```

Comment:
In Ruby, you can use the following syntax:
stream = destination || destination.catalog&.find_stream_by_name(stream_name)

Reply:
Added